### ISSUE TITLE
Inspired by Open Interpreter – Feedback on Code-Atlas

### ISSUE THREAD

**Issue by ystemsrx** [OpenInterpreter/open-interpreter#1538]
### Is your feature request related to a problem? Please describe.

I’ve been exploring Open Interpreter and I find its ability to process natural language and execute code incredibly innovative. Inspired by it, I developed a similar concept called [Code-Atlas](https://github.com/ystemsrx/Code-Atlas), which also allows natural language to control a computer. However, while my project is still in the early experimental stage, I am looking for feedback to improve it and possibly make it more effective in its functionality, particularly for resource-constrained devices.

### Describe the solution you'd like

I would like to refine my project by receiving feedback on how to better align it with the goals of Open Interpreter. Specifically, I’d like to know how I can improve its design, features (such as multi-language code execution and Markdown rendering), and general functionality to make it more efficient and user-friendly. Code-Atlas is implemented using C++ for better performance on devices with limited resources, so I’m particularly interested in optimizing the C++ implementation and ensuring cross-platform compatibility.

### Describe alternatives you've considered

I’ve considered adding more features such as additional programming language support and better integration with various operating systems. However, I haven’t implemented these yet because I want to first focus on optimizing the current version for lighter resource usage and stability.

### Additional context

I am currently a college student and Code-Atlas is just in its experimental, early stages. I’m particularly interested in learning from projects like Open Interpreter and exploring potential ways to enhance its capabilities. Any suggestions or advice would be greatly appreciated!

**Comment by Kreijstal** [comment]
ah very nice, I was looking for this, is there a way to use ollama?

**Comment by ystemsrx** [comment]
Yes, it fully supports Ollama. You just need to modify the config file, use the serve feature provided by Ollama, and set the `base_url` to `http://localhost:11434/api/chat`. This allows you to interact with the LLM through Ollama while also enabling you to adjust various parameters on it.

**Comment by Kreijstal** [comment]
> Yes, it fully supports Ollama. You just need to modify the config file, use the serve feature provided by Ollama, and set the `base_url` to `http://localhost:11434/api/chat`. This allows you to interact with the LLM through Ollama while also enabling you to adjust various parameters on it.

since you are using c++ you can use std::filesystem instead of win32 api for more portability, also consider using cmake, and curses for terminal stuff

**Comment by ystemsrx** [comment]
Thanks! I'll consider `std::filesystem` for better portability, and also look into CMake and curses. Really appreciate your support!

**Comment by interpreterwork** [comment]
Thank you for reporting this and for helping build the classic Python version of Open Interpreter. This issue concerns the earlier Python/LiteLLM/Jupyter product; this repository now tracks the Rust Open Interpreter built on Codex. We are closing the legacy report here so current product tracking starts clean. If you still want to pursue it, the community-maintained classic fork is https://github.com/endolith/open-interpreter. If the same problem affects the current Interpreter release, please open a new issue with the current template and include the version, provider, model, wire API, harness, platform, and reproduction.
